coordinator/src/main/scala/filodb.coordinator/queryplanner/LogicalPlanUtils.scala

```diff
@@ -92,14 +93,24 @@ object LogicalPlanUtils extends StrictLogging {
    */
   def copyLogicalPlanWithUpdatedTimeRange(logicalPlan: LogicalPlan,
                                           timeRange: TimeRange): LogicalPlan = {
+    // Snap the range endpoints to the previous second.
+    // Plans sometimes require (correctly) that their children share the same start/stop times,
+    //   but some plans do not support millisecond precision. A recursive time-range update might
+    //   cause two child plan time-ranges to mismatch if the range is not second-aligned.
+    // Given that millisecond precision is never necessary, timestamps are shifted left to the
+    //   nearest second in order to avoid the hassle.
+    val timeRangeToUse = TimeRange(
+      1000 * (timeRange.startMs / 1000),
+      1000 * (timeRange.endMs / 1000)
+    )
```

Comment:
> Given that millisecond precision is never necessary, ...

Anyone disagree?

Reply:
Are there other options?
1. Modify the timestamps here.
2. Remove/relax any assertions that check child plans have similar timestamps.
3. Add millisecond precision to all plans.

The check in (2) is correct and shouldn't be relaxed. (3) will require an update to all scalar plans/tests, and (1) is a good choice as long as:
> millisecond precision is never necessary